src/it/scala/com.rbmhtechnology.calliope/TransactionalEventProducerSpec.scala

```diff
@@ -229,16 +231,34 @@ class TransactionalEventProducerSpec extends KafkaSpec with MustMatchers with Ty
       "delete produced events from the underlying event store" in {
         val (topic, consumer) = topicConsumer()
         val eventStore = WritableEventStore()
-        val writer = runTransactionalEventProducer(topic, eventStore, settings.withDeleteInterval(500.millis))
+        val deleteInterval = 500.millis
+        val writer = runTransactionalEventProducer(topic, eventStore, settings.withReadBufferSize(10).withDeleteInterval(deleteInterval))
 
         writer.writeEvent(Event("1", "agg1"))
         writer.writeEvent(Event("2", "agg1"))
         writer.writeEvent(Event("3", "agg1"))
 
-        consumer.request(3)
-        consumer.expectNextN(3)
+        Thread.sleep((deleteInterval * 2).toMillis)
```

Comment:
I think the test fails with a lower timeout because the TEP is still starting while the first event-batch is written, so the (timeout * 1.5) is not long enough to run a first deletion cycle.
The first deletion cycle will only be started after the second `sleep` invocation thereby skipping the first deletion.
I will try to decrease the intervals to (timeout * 1.5) but introducing an initial delay to wait for the TEP to start.